letsencrypt_apache/display_ops.py

```diff
@@ -0,0 +1,76 @@
+"""Contains UI methods for Apache operations."""
+import os
+import zope.component
+
+from letsencrypt import interfaces
+
+import letsencrypt.display.util as display_util
+
+
+def select_vhost(domain, vhosts):
+    """Select an appropriate Apache Vhost.
+
+    :param vhosts: Available Apache Virtual Hosts
+    :type vhosts: :class:`list` of type `~obj.Vhost`
+
+    :returns: VirtualHost
+    :rtype: `~obj.Vhost`
+
+    """
+    if not vhosts:
+        return None
+    while True:
+        code, tag = _vhost_menu(domain, vhosts)
+        if code == display_util.HELP:
+            _more_info_vhost(vhosts[tag])
+        elif code == display_util.OK:
+            return vhosts[tag]
+        else:
+            return None
+
+
+def _vhost_menu(domain, vhosts):
+    """Select an appropriate Apache Vhost.
+
+    :param vhosts: Available Apache Virtual Hosts
+    :type vhosts: :class:`list` of type `~obj.Vhost`
+
+    :returns: Display tuple - ('code', tag')
+    :rtype: `tuple`
+
+    """
+    filename_size = 24
+    disp_name_size = 17
+    choices = []
+    for vhost in vhosts:
+        if len(vhost.names) == 1:
+            disp_name = next(iter(vhost.names))
+        elif len(vhost.names) == 0:
+            disp_name = ""
+        else:
+            disp_name = "Multiple Names"
+
+        choices.append(
+            "{0:{4}s} | {1:{5}s} | {2:5s} | {3:7s}".format(
```

Comment:
wow... could this be made simpler? maybe use `"...{fsize}...".format(fsize=filename_size)"`?